### PR TITLE
Changed from int to long int in Assembler

### PR DESCRIPTION
**Pull request by ChristophSmoch**
a couple of variables are now initialized as long int instead of int (for example triplet_count), to avoid overflowing.

### CODE REVIEW

**Comment by Huangzizhou** [comment]
It's very unlikely, but there's a [maximum size](https://stackoverflow.com/questions/3813124/c-vector-max-size) on std::vector, which can cause overflow. Probably worth checking max sizes of Eigen matrix and sparse matrix.

File: src/polyfem/assembler/Assembler.cpp [hunk 1]
```diff
@@ -178,7 +178,7 @@ namespace polyfem::assembler
 
 			auto storage = create_thread_storage(LocalThreadMatStorage(buffer_size, stiffness.rows(), stiffness.cols()));
 
-			const int n_bases = int(bases.size());
+			const long int n_bases = int(bases.size());
```
Comment (teseoch) [inline src/polyfem/assembler/Assembler.cpp:181]: this is ok

File: src/polyfem/assembler/Assembler.cpp [hunk 2]
```diff
@@ -203,7 +203,7 @@ namespace polyfem::assembler
 
 					assert(MAX_QUAD_POINTS == -1 || quadrature.weights.size() < MAX_QUAD_POINTS);
 					local_storage.da = vals.det.array() * quadrature.weights.array();
-					const int n_loc_bases = int(vals.basis_values.size());
+					const long int n_loc_bases = int(vals.basis_values.size());
```
Comment (teseoch) [inline src/polyfem/assembler/Assembler.cpp:206]: this one too

File: src/polyfem/assembler/Assembler.hpp [hunk 3]
```diff
@@ -36,11 +36,11 @@ namespace polyfem::assembler
 
 		virtual std::string name() const = 0;
 
-		int size() const { return size_; }
+		long int size() const { return size_; }
```
Comment (teseoch) [inline src/polyfem/assembler/Assembler.hpp:39]: there are probably ok, also changing the type and not where they are assigned probably dosent do anything

File: src/polyfem/assembler/Assembler.hpp [hunk 4]
```diff
@@ -36,11 +36,11 @@ namespace polyfem::assembler
...
 		virtual void set_size(const int size) { size_ = size; }
 
 	protected:
-		int size_ = -1;
+		long int size_ = -1;
```
Comment (teseoch) [inline src/polyfem/assembler/Assembler.hpp:43]: int should be fine

File: src/polyfem/assembler/Assembler.hpp [hunk 5]
```diff
@@ -59,7 +59,7 @@ namespace polyfem::assembler
 
 		virtual std::string name() const = 0;
 
-		int size() const { return size_; }
+		long int size() const { return size_; }
```
Comment (teseoch) [inline src/polyfem/assembler/Assembler.hpp:62]: here too

File: src/polyfem/assembler/Assembler.hpp [hunk 6]
```diff
@@ -219,7 +219,7 @@ namespace polyfem::assembler
 		virtual bool is_tensor() const { return false; }
 
 	protected:
-		int size_ = -1;
+		long int size_ = -1;
```
Comment (teseoch) [inline src/polyfem/assembler/Assembler.hpp:222]: int?

**Pull request merged**